lib/view_component/base.rb

```diff
@@ -158,7 +158,7 @@ def controller
     # @return [ActionView::Base]
     def helpers
       raise ViewContextCalledBeforeRenderError, "`helpers` can only be called at render time." if view_context.nil?
-      @helpers ||= controller.view_context
+      @helpers ||= view_context
```

Comment:
Making this change resulted in a test error:

```
Error:
ViewComponentTest#test_renders_helper_method_within_nested_component:
NoMethodError: undefined method `message' for #<ContainerComponent:0x00007fd6ac718d08>
    /Users/cheshire137/code/github/view_component/test/app/components/helpers_proxy_component.html.erb:1:in `call'
```

I made the test pass by turning it into an integration test in test/view_component/integration_test.rb instead.